packages/esm-generic-patient-widgets-app/src/obs-table/obs-table.component.tsx

```diff
@@ -23,8 +23,8 @@ const ObsTable: React.FC<ObsTableProps> = ({ patientUuid }) => {
   const { t } = useTranslation();
   const config = useConfig();
   const { data: obss, error, isLoading, isValidating } = useObs(patientUuid, config.showEncounterType);
-  const uniqueDates = [...new Set(obss.map((o) => o.issued))].sort();
-  const obssByDate = uniqueDates.map((date) => obss.filter((o) => o.issued === date));
+  const uniqueDates = [...new Set(obss.map((o) => o.encounter.reference.split('#')[1]))].sort();
```

Comment:
I'd like @ibacher's input on whether this is a legal way to get a date

Reply:
The [HL7 docs](https://hl7.org/fhir/references-definitions.html#Reference.reference) don't specify much about the actual format of the reference, AFAICT

Reply:
I think this doesn't do what it's supposed to do. References are, essentially, either URLs or URL parts so something like:

https://example.org/fhir/Encounter/8081f1a1-8c64-495f-b21c-f7395484a705
Encounter/8081f1a1-8c64-495f-b21c-f7395484a705

It can also contain ids using the `#`, but it's not a standard part.

What this definitely doesn't do is extract a date or time at all, unless there's some kind of pre-processing I'm unaware of.

Reply:
Sorry, that's my bad, it's actually returning the encounter UUIDs, instead of dates, I didn't update the variable name. Let me just fix that.

Reply:
From the Observation response, we get the reference of the Encounter in the form of `Encounter#<Associated encounter UUID>`.

Reply:
Also, I don't know why I used `#`, but it's actually `/`, thanks @ibacher for the catch.
I am not splitting it anymore and using the whole encounter reference for grouping.
Thanks!

Reply:
Pushed the latest change!